docs/architecture-cli.md

```diff
@@ -0,0 +1,133 @@
+---
+title: The Talk cli
+keywords: architecture
+sidebar: talk_sidebar
+permalink: architecture-cli.html
+summary:
+---
+
+Talk ships with a cli tool that allows access to a wide variety of functionality.
+
+It is designed to provide a convenient way for engineers to perform key tasks without the need to muck about in the UI. It also opens the door for scripts to perform operations programmatically.
+
+Note: the cli tool requires [the Talk environment to be configured](configuration.html) either via env vars or by using a `.cli` file via `bin/cli -c .env [command] ....`
+
+## Using the cli
+
+In a terminal, try:
+
+```
+/path/to/talk/bin cli [options] [commands] [arguments]
+```
+
+Commonly, you'll be in the `talk/` folder, in which case you can:
+
+```
+bin cli [options] [commands] [arguments]
+```
+
+If you're a heavy cli user, consider adding the `bin` folder to your path so you can run `cli` from anywhere!
+
+## What can I do with the cli?
+
+The Talk cli ships with 'unix style' help. To access the docs, simply run the cli with insufficient arguments.
+
+Let's say I wanted to figure out how to change a user's password. I'd start be seeing what the cli has for me.
+
+```
+talk :) ]$ bin/cli
+
+  Usage: cli [options] [command]
+
+
+  Commands:
+
+    serve       serve the application
+    settings    interact with the application settings
+    assets      interact with assets
+    setup       setup the application
+    jobs        work with the job queues
+    token       work with the access tokens
+    users       work with the application auth
+    migration   provides utilities for migrating the database
+    plugins     provides utilities for interacting with the plugin system
+    help [cmd]  display help for [cmd]
+
+  Options:
+
+    -h, --help           output usage information
+    -V, --version        output the version number
+    -c, --config [path]  Specify the configuration file to load
+    --pid [path]         Specify a path to output the current PID to
+```
```

Comment:
I'll add a note that this can change. One of the mindsets that I'm trying to develop here is that people _think in terms of the cli_. Having this verbose output here I believe will trigger folks to think *woah, there's a lot I can do* and actually try it for themselves. This is under the flag of showing off/selling what we have to people who may not have tried it yet.